interview-questions.md

```diff
@@ -8,50 +8,56 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: a parameter is the value of an object, the index and the array you are performing a the function on. while an argument is the action of what you would like the output to return.
 
-Researched answer:
+Researched answer: A parameter is a variable that has been given a name and passed through into a function as "placeholder" The argument is the actual value of an object or array that the function will run actions on.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: the built in method .map () uses the parameters value,index, and the array. The optional predefined parameters that are optional are the index and array.
 
-Researched answer:
+Researched answer: The built in method .map will require the current value, the index and the higher order function that an array is called on but only the value of the array at the current iteration is required. 
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: . map() is used to take a value in a function that has an array and returns a new array with the value of the return requested. . filter() is designed to take a conditional statement on on a function with an array and return the specific response to the conditional statement. 
 
-Researched answer:
+Researched answer: . map() will return each value of an array within a new array, based on the return action requested. for example if .map() was used to find the multiple of 3 in an array of const myArray = [4,7,9,11]
+
+const multiplier = myArray.map((value) => {
+  return value  3 % 3 === 0
+})
+
+while .filter() returns the output of a conditional statement narrowing down to more specific values such as returning the value at index 3.
 
 1. What is iteration?
 
-Your answer:
+Your answer: The constant running of a function by the terminal based on parameters given to return a desire outcome.
 
-Researched answer:
+Researched answer: Iteration is the protocol that runs the code to the next operator to return a value or true / false response.
```

Comment:
Iteration is performing something on each element in a collection. For example, with .map we can multiply the values in an array by a number. Going into each element is the iteration.